actions/post_actions.jsx

```diff
@@ -56,57 +56,64 @@ export function handleNewPost(post, msg) {
     };
 }
 
-export async function flagPost(postId) {
-    await PostActions.flagPost(postId)(dispatch, getState);
-
-    const rhsState = getRhsState(getState());
-
-    // This is a hack that should be fixed with better reducers/actions, see MM-9793
-    if (rhsState === RHSStates.FLAG) {
-        let results = getState().entities.search.results;
-        const index = results.indexOf(postId);
-        if (index === -1) {
-            results = [...results, postId];
-
-            const posts = {};
-            results.forEach((id) => {
-                posts[id] = PostSelectors.getPost(getState(), id);
-            });
-
-            results.sort((a, b) => comparePosts(posts[a], posts[b]));
-
-            dispatch({
-                type: SearchTypes.RECEIVED_SEARCH_POSTS,
-                data: {posts, order: results},
-            });
+export function flagPost(postId) {
+    return async (doDispatch, doGetState) => {
+        await doDispatch(PostActions.flagPost(postId));
+        const state = doGetState();
+        const rhsState = getRhsState(state);
+
+        if (rhsState === RHSStates.FLAG) {
+            const results = state.entities.search.results;
+            const index = results.indexOf(postId);
+            if (index === -1) {
+                const flaggedPost = PostSelectors.getPost(state, postId);
+
+                const getPostsForIds = PostSelectors.makeGetPostsForIds();
```

Comment:
Should the variable `getPostsForIds` be instantiated on every invocation of `flagPost` or should it be reused from an outer scope?

Reply:
Good point. I instantiated the `getPostsForIds` on outer scope and have it reusable.  Thanks @mkraft!